addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,152 @@
+        error_res = (response.get('error') and response) or default_error_msg
+        return error_res
+
+    def paytm_fetch_payment_status(self, transaction_id, reference_id, timestamp):
+        body = self._paytm_get_request_body(transaction_id, reference_id, timestamp)
+        head = self._paytm_get_request_head(body)
+        payload = {'head': head, 'body': body}
+        response = self._paytm_make_request('V2/payment/status', payload=payload)
+        result_code = response.get('resultInfo', {}).get('resultCode')
+        if result_code == 'S':
+            # Since we don't want to send extra data on RPC call
+            # Only sending essential data when the transaction is successful
+            data = response['resultInfo']
+            data.update({
+                    'authCode': response.get('authCode'),
+                    'issuerMaskCardNo': response.get('issuerMaskCardNo'),
+                    'issuingBankName': response.get('issuingBankName'),
+                    'payMethod': response.get('payMethod'),
+                    'cardType': response.get('cardType'),
+                    'cardScheme': response.get('cardScheme'),
+                    'merchantReferenceNo': response.get('merchantReferenceNo'),
+                    'merchantTransactionId': response.get('merchantTransactionId'),
+                    'transactionDateTime': response.get('transactionDateTime'),
+            })
+            return data
+        elif result_code == 'F':
+            return {'error': "%s" % response['resultInfo'].get('resultMsg', _('paytm transaction failure'))}
+        elif result_code == 'P':
+            return response['resultInfo']
+        default_error_msg = {'error': _('paymentFetchRequest expected resultCode not found in the response')}
+        error_res = (response.get('error') and response) or default_error_msg
+        return error_res
+
+    def _paytm_generate_signature(self, params_dict, key):
+        params_list = []
+        for k in sorted(params_dict.keys()):
+            value = params_dict[k]
+            if value is None or params_dict[k].lower() == "null":
+                value = ""
+            params_list.append(str(value))
+        salt = ''.join(secrets.choice(string.ascii_letters + string.digits) for _ in range(4))
+        params_list.append(salt)
+        params_with_salt = '|'.join(params_list)
+        hashed_params = hashlib.sha256(params_with_salt.encode())
+        hashed_params_with_salt = hashed_params.hexdigest() + salt
+        padding = 12 #the padding value is a constant
+        padded_hashed_params_with_salt = bytes(hashed_params_with_salt + padding * chr(padding), 'utf-8')
+        cipher = Cipher(algorithms.AES(key.encode()), modes.CBC(iv))
+        encryptor = cipher.encryptor()
+        encrypted_hashed_params = encryptor.update(padded_hashed_params_with_salt) + encryptor.finalize()
+        return base64.b64encode(encrypted_hashed_params).decode("UTF-8")
```

Comment:
add a try catch and forward the error.
Because if the customer encode a wrong or too long/short credential you get a error 
And the cashier did not know why